internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -558,3 +561,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	cr.SetConditions(xpv1.Available())
 	return reconcile.Result{RequeueAfter: longWait}, errors.Wrap(r.client.Status().Update(ctx, cr), errUpdateStatus)
 }
+
+func filterToXRPatches(tas []TemplateAssociation) []v1.Patch {
+	filtered := make([]v1.Patch, 0, len(tas))
```

Comment:
`len(tas)` is not an upper bound for the number of elements in the result to be returned, but linter asked me pre-allocate this array and I think it's better than initialing with 0 capacity.